DOCS-2210: Scale Cloud Sync Article (#943)
truenas/documentation: content/SCALE/DataProtection/CloudSyncTasks.md

---
title: "Cloud Sync Tasks"
weight: 10
---

{{< toc >}}

> TrueNAS can send, receive, or synchronize data with a Cloud Storage provider.  Cloud Sync tasks allow for single time transfers or recurring transfers on a schedule, and are an effective method to back up data to a remote location.

Reviewer: ```suggestion
TrueNAS can send, receive, or synchronize data with a Cloud Storage provider. Cloud Sync tasks allow for single time transfers or recurring transfers on a schedule and are an effective method to back up data to a remote location.
```

{{< hint warning >}}

> Using the Cloud means that data can go to a third party commercial vendor not directly affiliated with iXsystems.  Please investigate and fully understand that vendor’s pricing policies and services before creating any Cloud Sync task. iXsystems is not responsible for any charges incurred from the use of third party vendors with the Cloud Sync feature.

Reviewer: ```suggestion
Using the Cloud means that data can go to a third party commercial vendor not directly affiliated with iXsystems. Please investigate and fully understand that vendor’s pricing policies and services before creating any Cloud Sync task. iXsystems is not responsible for any charges incurred from the use of third party vendors with the Cloud Sync feature.
```

{{< /hint >}}

> TrueNAS supports major providers like Amazon S3, Google Cloud, and Microsoft Azure, along with a variety of other vendors.  To see the full list of supported vendors, go to **Credentials > Backup Credentials > Cloud Credentials** click *Add* and open the *Provider* dropdown.

Reviewer: ```suggestion
TrueNAS supports major providers like Amazon S3, Google Cloud, and Microsoft Azure, along with a variety of other vendors. To see the full list of supported vendors, go to **Credentials > Backup Credentials > Cloud Credentials** click *Add* and open the *Provider* dropdown.
```

## Requirements

* All system [Storage](/SCALE/storage/) must be configured and ready to receive or send data.

> * A Cloud Storage provider account and a cloud storage location must be available, like an Amazon S3 bucket.

Reviewer: ```suggestion
* A Cloud Storage provider account and a cloud storage location, like an Amazon S3 bucket, must be available.
```

{{< /tab >}}
{{< /tabs >}}

Next, **Control** when the task runs by defining a *Schedule*. When a specific *Schedule* is required, choose *Custom* and use the **Advanced Scheduler**.

{{< expand "Advanced Scheduler" "v" >}}
{{< include file="static/includes/AdvancedScheduler.md.part" markdown="true" >}}
{{< /expand >}}

Unsetting *Enable* makes the configuration available without allowing the *Schedule* to run the task. To manually activate a saved task, go to **Data Protection > Cloud Sync Tasks**, click <i class="fa fa-play" aria-hidden="true"></i> for the cloud sync task you want to run.  You will be prompted to *CONTINUE* or *CANCEL* the *Run Now* operation.

The remaining options allow tuning the task to your specific requirements.

{{< expand "Specific Options" "v" >}}
{{< include file="static/includes/TasksCloudSyncAddFields.md.part" markdown="true" >}}

### Scripting and Environment Variables

Advanced users can write scripts that run immediately *before* or *after* the Cloud Sync task. The **Post-script** field is only run when the Cloud Sync task successfully completes. You can pass a variety of task environment variables into the **Pre-** and **Post-** script fields:

> * CLOUD_SYNC_ID
> * CLOUD_SYNC_DESCRIPTION
> * CLOUD_SYNC_DIRECTION
> * CLOUD_SYNC_TRANSFER_MODE
> * CLOUD_SYNC_ENCRYPTION
> * CLOUD_SYNC_FILENAME_ENCRYPTION
> * CLOUD_SYNC_ENCRYPTION_PASSWORD
> * CLOUD_SYNC_ENCRYPTION_SALT
> * CLOUD_SYNC_SNAPSHOT

Reviewer: ```suggestion
* `CLOUD_SYNC_ID`
* `CLOUD_SYNC_DESCRIPTION`
* `CLOUD_SYNC_DIRECTION`
* `CLOUD_SYNC_TRANSFER_MODE`
* `CLOUD_SYNC_ENCRYPTION`
* `CLOUD_SYNC_FILENAME_ENCRYPTION`
* `CLOUD_SYNC_ENCRYPTION_PASSWORD`
* `CLOUD_SYNC_ENCRYPTION_SALT`
* `CLOUD_SYNC_SNAPSHOT`
```

There also are provider-specific variables like CLOUD_SYNC_CLIENT_ID or CLOUD_SYNC_TOKEN or CLOUD_SYNC_CHUNK_SIZE

Remote storage settings:

> * CLOUD_SYNC_BUCKET
> * CLOUD_SYNC_FOLDER

Reviewer: ```suggestion
* `CLOUD_SYNC_BUCKET`
* `CLOUD_SYNC_FOLDER`
```

Local storage settings:

> * CLOUD_SYNC_PATH

Reviewer: ```suggestion
* `CLOUD_SYNC_PATH`
```

{{< /expand >}}

### Testing Settings

> Test the settings before saving by clicking *Dry Run*. TrueNAS connects to the Cloud Storage Provider and simulates a file transfer. No data is actually sent or received.
> A dialog shows the test status and allows downloading the task logs.  This can also be ran by clicking <i class="material-icons" aria-hidden="true" title="Dry Run">loop</i> after the task has been created.

Reviewer: ```suggestion
Test the settings before saving by clicking *Dry Run*. TrueNAS connects to the Cloud Storage Provider and simulates a file transfer. No data is actually sent or received.
A dialog shows the test status and allows downloading the task logs. This is also ran by clicking <i class="material-icons" aria-hidden="true" title="Dry Run">loop</i> after the task is created.
```

![DataProtectionCloudSyncAddBoxDryRun](/images/SCALE/DataProtectionCloudSyncDryRun.png "Example: Box Drive Test")

## Cloud Sync Behavior

> Saved tasks are activated according to their schedule or by clicking <i class="material-icons" aria-hidden="true" title="Run Now">play_arrow</i>. An in-progress cloud sync must finish before another can begin. Stopping an in-progress task cancels the file transfer and requires starting the file transfer over.

Reviewer: ```suggestion
Saved tasks activate according to their schedule or by clicking <i class="material-icons" aria-hidden="true" title="Run Now">play_arrow</i>. An in-progress cloud sync must finish before another can begin. Stopping an in-progress task cancels the file transfer and requires starting the file transfer over.
```

To view logs about a running or the most recent run of a task, click the task *State*.

## Cloud Sync Restore

To quickly create a new Cloud Sync that uses the same options but reverses the data transfer, select <i class="material-icons" aria-hidden="true" title="Restore">history</i> for an existing Cloud Sync on the **Data Protection** page.

![DataProtectionsCloudSyncRestore](/images/SCALE/DataProtectionCloudSyncRestore.png "Cloud Sync Restore")

Enter a new *Description* for this reversed task, define the path to a storage location for the transferred data and click *Restore*.

The restored cloud sync is saved as another entry in **Data protection > Cloud Sync Tasks**.

> In case the restore destination dataset is the same as the original source dataset, the restored files might have their ownership altered to *root*. If the original files were not created by *root* and a different owner is required, you can recursively reset ACL Permissions of the restored dataset through the GUI or by running `chown` from the CLI.

Reviewer: ```suggestion
In case the restore destination dataset is the same as the original source dataset, the restored files might have their ownership altered to *root*. If the original files were not created by *root* and a different owner is required, you can recursively reset [ACL Permissions]({{< relref "Permissions.md" >}}) of the restored dataset through the GUI or by running `chown` from the CLI.
```
